packages/live-share/src/EphemeralObjectSynchronizer.ts

```diff
@@ -226,24 +230,58 @@ class ContainerSynchronizer {
                 this._hTimer = undefined;
                 return true;
             }
+
+            // Remove id from key lists
+            this._connectedKeys = this._connectedKeys.filter(key => key != id);
+            this._unconnectedKeys = this._unconnectedKeys.filter(key => key != id);
         }
 
         return false;
     }
 
-    private sendUpdateEvent(keys: string[]): void {
+    private sendGroupEvent(keys: string[], evt: string): void {
         // Compose list of updates
+        let send = false;
         const updates: StateSyncEventContent = {};
         keys.forEach((id) => {
             try {
                 const state = this._objects.get(id)?.getState(false);
-                updates[id] = state;
+                if (typeof state == 'object') {
+                    updates[id] = state;
+                    send = true;
+                }
             } catch (err: any) {
                 console.error(`EphemeralObjectSynchronizer: error getting an objects state - ${err.toString()}`);
             }
         });
 
-        // Send update event
-        this._containerRuntime.submitSignal(UPDATE_EVENT, updates);
+        // Send event if we have any updates to broadcast
+        if (send) {
```

Comment:
instead of tracking a boolean, could rely on the updates itself? could maybe be slower
```typescript
if (Object.keys(updates).length > 0) {
    ...
}
```

Reply:
Good idea... It will be a little slower but it's one less potential bug so I'll make the change.